java/com/google/cose/exceptions/CoseException.java

```diff
@@ -0,0 +1,15 @@
+package com.google.cose.exceptions;
+
+/**
+ * This class is used for throwing any exceptions that may occur during parsing and serialization of
+ * byte data into their respective COSE components.
+ */
+public class CoseException extends RuntimeException {
```

Comment:
Are we sure this is a `RuntimeException`? It seems like we might want this checked, as we're going to throw on bad data. If callers aren't made explicitly aware of the exception, then they may not catch. Bad data crashing our services seems unwanted.

Reply:
Updated this to a checked exception now.